api/admin/group.py

```diff
@@ -0,0 +1,174 @@
+            "id",
+            "language_and_level",
+            "coordinators",
+            "teachers",
+            "students",
+            "start_date",
+            "end_date",
+            "project_status",
+            "situational_status",
+            "status_since",
+            "is_for_staff_only",
+        )
+
+
+class GroupAdminCustom(VersionAdmin):
+    form = GroupAdminForm
+    list_display = (
+        "id",
+        "language_and_level",
+        "coordinators_list",
+        "teachers_list",
+        "students_count",
+        "start_date",
+        "end_date",
+        "get_schedule",
+        "project_status",
+        "situational_status",
+        "status_since",
+        "staff_only",
+    )
+
+    list_filter = (
+        StuffOnlyFilter,
+        "project_status",
+        "situational_status",
+        TeachersFilter,
+        CoordinatorsFilter,
+    )
+
+    search_fields = (
+        "coordinators__personal_info__first_name",
+        "coordinators__personal_info__last_name",
+        "teachers__personal_info__first_name",
+        "teachers__personal_info__last_name",
+    )
+
+    def coordinators_list(self, obj: models.Group) -> str:
+        links = [
+            format_html(
+                '<a href="{}">{}</a>',
+                reverse("admin:api_coordinator_change", args=(coordinator.pk,)),
+                coordinator.personal_info.full_name,
+            )
+            for coordinator in obj.coordinators.all()
+        ]
+        return mark_safe(", ".join(links))
+
+    coordinators_list.short_description = "Coordinators"  # type: ignore
+
+    def teachers_list(self, obj: models.Group) -> str:
+        links = [
+            format_html(
+                '<a href="{}">{}</a>',
+                reverse("admin:api_teacher_change", args=(teacher.pk,)),
+                teacher.personal_info.full_name,
+            )
+            for teacher in obj.teachers.all()
+        ]
+        return mark_safe(", ".join(links))
+
+    teachers_list.short_description = "Teachers"  # type: ignore
+
+    def students_count(self, obj: models.Group) -> int:
+        return obj.students.count()
+
+    students_count.short_description = "Number of students"  # type: ignore
+
+    def get_schedule(self, obj: models.Group) -> str:
+        day_names = ["Mo", "Tu", "We", "Th", "Fr", "Sa", "Su"]
+        days = ["monday", "tuesday", "wednesday", "thursday", "friday", "saturday", "sunday"]
+        schedule = [
+            f"{day_names[i]}: {getattr(obj, day).strftime('%H:%M')}"
+            for i, day in enumerate(days)
+            if getattr(obj, day)
+        ]
+        return mark_safe(",<br>".join(schedule))
+
+    get_schedule.short_description = "Schedule"  # type: ignore
+    get_schedule.allow_tags = True  # type: ignore
+
+    def staff_only(self, obj: models.Group) -> str:
+        if getattr(obj, "is_for_staff_only"):
+            return format_html('<img src="{}" alt="icon"/>', "/static/admin/img/icon-yes.svg")
```

Comment:
Why not just keep "is_for_staff_only", in fields? It will show the same icon for boolean field

Reply:
If I understood correctly request was to have green icon for staff members and to display no icon in cases there are no staff members

Reply:
Strange request :)But ok, let's keep it